src/Compilers/CSharp/Portable/Parser/LanguageParser_InterpolatedString.cs

```diff
@@ -147,46 +134,76 @@ private ExpressionSyntax ParseInterpolatedStringToken()
             return CheckFeatureAvailability(result, MessageID.IDS_FeatureInterpolatedStrings);
         }
 
+        private static InterpolationSyntax ParseInterpolation(CSharpParseOptions options, string text, Lexer.Interpolation interpolation, bool isVerbatim)
+        {
+            // Grab from before the { all the way to the start of the }.  The parsing of the close curly is specially
+            // handled in ParseInterpolation below.
+            var parsedText = Substring(text, TextSpan.FromBounds(
+                interpolation.OpenBraceSpan.Start,
+                interpolation.HasColon ? interpolation.ColonSpan.Start : interpolation.CloseBraceSpan.Start));
+
+            // TODO: some of the trivia in the interpolation maybe should be trailing trivia of the openBraceToken
+            using var tempLexer = new Lexer(SourceText.From(parsedText), options, allowPreprocessorDirectives: false, interpolationFollowedByColon: interpolation.HasColon);
+            using var tempParser = new LanguageParser(tempLexer, oldTree: null, changes: null);
+
+            return tempParser.ParseInterpolation(text, interpolation, isVerbatim);
+        }
+
         private InterpolationSyntax ParseInterpolation(string text, Lexer.Interpolation interpolation, bool isVerbatim)
         {
-            SyntaxToken openBraceToken;
-            ExpressionSyntax expression;
-            InterpolationAlignmentClauseSyntax alignment = null;
-            InterpolationFormatClauseSyntax format = null;
-            var closeBraceToken = interpolation.CloseBraceMissing
-                ? SyntaxFactory.MissingToken(SyntaxKind.CloseBraceToken)
-                : SyntaxFactory.Token(SyntaxKind.CloseBraceToken);
-
-            var parsedText = Substring(text, interpolation.OpenBracePosition, interpolation.HasColon ? interpolation.ColonPosition - 1 : interpolation.CloseBracePosition - 1);
-            using (var tempLexer = new Lexer(Text.SourceText.From(parsedText), this.Options, allowPreprocessorDirectives: false, interpolationFollowedByColon: interpolation.HasColon))
+            var openBraceToken = this.EatToken(SyntaxKind.OpenBraceToken);
+            var (expression, alignment) = getExpressionAndAlignment();
+            var (format, closeBraceToken) = getFormatAndCloseBrace();
```

Comment:
complex interpolation parsing was broken into it's three successive segments.  The parts taht return pairs of items do so as there's a relation between those two pieces (esp. wrt to how trivia is associated with either part).